File: Code C++/Invoice/Invoice.cpp
//San Diego - 28/01/2018 - 16:07.
//Definindo as Funções da classe Invoice.
#include <iostream>
#include <string>
#include "Invoice.h"

using std::cout;
using std::endl;
using std::string;

//Construtor, resposável por inicializar os valores dos objetos da classe Invoice.
Invoice::Invoice(string ide, string desc, int amo, int pri){
	//guarda os respectivos valores.
	setIdentifier(ide);
	setDescription(desc);
	setAmount(amo);
	setPrice(pri);
}

//Função responsável por guardar o identificador de um objeto da classe Invoice.
void Invoice::setIdentifier(string ide){
	identifier = ide;
}

//Função resposável por retornar o identificador de um objeto da classe Invoice.
string Invoice::getIdentifier(){
	return identifier;
}

//Função responsável por guardar a descrição de um objeto da classe Invoice.
void Invoice::setDescription(string desc){
	description = desc;
}

//Função responsável por retornar a descrição de um objeto da classe Invoice.
string Invoice::getDescription(){
	return description;
}

//Função resposável por guardar a quantidade de itens comprados.
void Invoice::setAmount(int amo){

	if (amo < 0)
		amount = 0;

	amount = amo;
}

//Função resposável por retornar a quantidade de itens comprados.
int Invoice::getAmount(){
	return amount;
}

//Função resposável por guardar o valor de cada item comprado.
void Invoice::setPrice(int pri){

	if (pri < 0)
		price = 0;

	price = pri;
}

//Função responsável por retorar o valor de cada item comprado.
int Invoice::getPrice(){
	return price;
}

//Função responsável por calcular o valor total da fatura e retornar esse valor.
int Invoice::getInvoiceAmount(){
	int result = price * amount;

	return result;
}

File: Learning JAVA/Capítulo 03/Invoice/Invoice.java
//San Diego - 24/05/2020
// Classe que representa a fatura de uma loja de informática

public class Invoice{

	//Variáveis de instância
	private String number;
	private String description;
	private int amount;
	private double price;

	//Construtor
	public Invoice (String number, String description, int amount, double price){

		this.number = number;
		this.description = description;

		if (amount > 0 && price > 0.0) {
			this.amount = amount;
			this.price  = price;
		}
	}


	//Getters e Setters dos atributos
	public void setNumber(String number){
		this.number = number;
	}

	public String getNumber(){
		return number;
	}

	public void setDescription(String description){
		this.description = description;
	}

	public String getDescription(){
		return description;
	}

	public void setAmount(int amount){

		if (amount > 0)
			this.amount = amount;
	}

	public int getAmount(){
		return amount;
	}

	public void setPrice(double price){

		if (price > 0.0)
			this.price = price;
	}

	public double getPrice(){
		return price;
	}

	//Calcula o valor total da fatura
	public double getInvoiceAmount(){
		double invoiceAmount;

		invoiceAmount = (getPrice() * getAmount());

		return invoiceAmount;
	}

}